addons/website_forum/populate/forum_post.py

```diff
@@ -0,0 +1,28 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from random import choices
+from odoo import models, Command
+from odoo.tools import populate
+
+QA_WEIGHTS = {0: 25, 1: 35, 2: 20, 3: 10, 4: 4, 5: 3, 6: 2, 7: 1}
+
+class Post(models.Model):
+    _inherit ='forum.post'
+    _populate_sizes = { 'small': 100, 'medium': 1000, 'large': 90000 }
+    _populate_dependencies = ['forum.forum']
+
+    def _populate_factories(self):
+        forum_ids = self.env.registry.populated_models['forum.forum']
+
+        def create_answers(iterator, *args):
+            for values in iterator:
+                answers = [Command.create({'name': f"reply to {values['name']}", 'forum_id': values['forum_id']})
+                            for _ in range(choices(*zip(*QA_WEIGHTS.items()))[0])]
```

Comment:
I would split this into multiple lines to make it more readable. Also, a small comment can explain that we need this "complex" mechanism because answers need the forum_id even thought the parent could logically be enough.
```suggestion
                answers = [
                    Command.create({
                        'name': f"reply to {values['name']}", 
                        'forum_id': values['forum_id']}
                    ) for _ in range(choices(*zip(*QA_WEIGHTS.items()))[0])
                ]
```